instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/test/java/io/opentelemetry/test/annotation/SpanAttributesWithCurrentSpan.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.test.annotation;
+
+import io.opentelemetry.instrumentation.annotations.SpanAttribute;
+import io.opentelemetry.instrumentation.annotations.WithCurrentSpan;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+
+public class SpanAttributesWithCurrentSpan {
+
+  @WithCurrentSpan
+  public String withCurrentSpanAttributes(
+      @SpanAttribute String implicitName,
+      @SpanAttribute("explicitName") String parameter,
+      @SpanAttribute("nullAttribute") String nullAttribute,
+      String notTraced) {
+
+    return "hello!";
+  }
+
+  @WithSpan
+  @WithCurrentSpan
+  public String withSpanAttributes(
```

Comment:
I think a reader would quickly question this use of seemingly conflicting annotations. One explicitly creates a new span, one explicitly does not and uses the parent context. You can't have it both ways, so which one wins? It's not easily understood.

Looking at the test, it seems like `@WithCurrentSpan` wins here, because there's only 1 span in the resulting trace. Can you confirm that that is intentional?

Reply:
Yes, that is intentional. The WithCurrenSpan instrumentation explicitly filters out any method that happens for some reason to be annotated with both.

Reply:
```
isAnnotatedWith(
                named("application.io.opentelemetry.instrumentation.annotations.WithCurrentSpan"))
            // Avoid repeat extraction if method is already annotation with WithSpan
            .and(
                not(
                    isAnnotatedWith(
                        named(
                            "application.io.opentelemetry.instrumentation.annotations.WithSpan"))));
```

Reply:
It will overwrite in the same way calls to `span.setAttribute(k,v) would if the k was already set.